rust/gui-client/src-common/src/controller.rs

```diff
@@ -143,6 +188,33 @@ impl Status {
 
 impl<I: GuiIntegration> Controller<I> {
     pub async fn main_loop(mut self) -> Result<(), Error> {
+        // Ask for user consent for telemetry if we haven't asked before
+        if let Some(enable_telemetry) = self.advanced_settings.enable_telemetry {
+            tracing::info!(?enable_telemetry, "Not showing telemetry consent dialog");
+            self.telemetry
+                .set_enabled(enable_telemetry.then_some(telemetry::GUI_DSN));
+        } else {
+            let enable_telemetry = rfd::AsyncMessageDialog::new()
+                .set_buttons(rfd::MessageButtons::YesNo)
+                .set_description("Enable sentry.io telemetry? This is completely optional, but it helps us debug Firezone faster.")
```

Comment:
```suggestion
                .set_description("Enable sentry.io telemetry? This helps us make Firezone better.")
```

YMMV, I don't think we need to spell out that telemetry is optional.